src/ValueFormatters/QuantityUnitFormatter.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace ValueFormatters;
+
+/**
+ * Interface defining a service for formatting a quantity's unit.
+ *
+ * @licence GNU GPL v2+
+ * @author Daniel Kinzler
+ */
+interface QuantityUnitFormatter {
+
+	/**
+	 * Attaches a human readable version of the unit to the given number text.
+	 * The number text is expected to already be formatted, and is not modified.
+	 *
+	 * The unit may be placed before or after the number, with or without a separator.
+	 * The unit may also be omitted in case the $unit represents "no unit".
+	 *
+	 * @param string $unit
+	 * @param string $numberText
+	 *
+	 * @return string A text representing the given number with the given unit applied.
+	 */
+	public function applyUnit( $unit, $numberText );
```

Comment:
I just find the name odd. A number is a number. Text is text. Moreover, this code does not care what the string contains. Which makes me think why the string is passed to this class in the first place. I think the responsibility of concatenating the two strings should not be here but in the QuantityFormatter. So I suggest to change this class into a UnitFormatter with a format method that gets a string and returns a string.